src/v2/Apps/Artwork/ArtworkApp.tsx

```diff
@@ -223,6 +230,9 @@ const TrackingWrappedArtworkApp: React.FC<Props> = props => {
   // Check to see if referrer comes from link interception.
   // @see interceptLinks.ts
   const referrer = state && state.previousHref
+  const submittedOrder = state && state.submittedOrder
+  // Discard the submitted order from the state after passing it the first time
+  if (submittedOrder) state.submittedOrder = null
```

Comment:
>Would it be that the case if line 233 is a boolean coming from the review page? I guess the piece I'm missing is how can the artwork page know if the user comes from a specific use case (MOOAEA)
See [these lines on a previous PR](https://github.com/artsy/force/pull/9241/files#diff-1aaa2033a9ae43ffb3da1495105eac4324ed37ff31bbf19c13da79955ac4da47R151-R154) as an example.

Yes, I guess it makes sense, I would also just checking returning to the specific artwork after the redirect to the conversations to make sure that state is not persisted causing issues when collectors coming back to artworks

Reply:
> I would also just checking returning to the specific artwork after the redirect to the conversations to make sure that state is not persisted causing issues when collectors coming back to artworks

That's what emptying the state variable was achieving. I get it's an anti-pattern, though. Any ideas on how to achieve the same? Maybe passing a variable to the router state is not the right option? Maybe the same approach but breaking fewer rules would be passing a query string on the artwork page URL instead. wdyt?

Reply:
I guess another problem is how the artwork page can know for sure which order belongs to it. I can query `me.orders` and check the `artworks` there to match. Not sure what happens if, for example, a collector makes more than an offer on the same artwork, though (maybe on different editions).

Reply:
I'm fine with the url param to be very honest, I prefer it since we don't need to persist anything

>I guess another problem is how the artwork page can know for sure which order belongs to it. I can query me.orders and check the artworks there to match. Not sure what happens if, for example, a collector makes more than an offer on the same artwork, though (maybe on different editions).

I guess you have something like myLastOffer to query that grabs only the last one

Reply:
for uh-oh edition sets if it's easy we can handle it here, otherwise, we ticket a follow up on it

Reply:
Follow-up on this conversation: I pushed changes moments ago that I'll keep working on, but wanted to quickly check with you @araujobarret. It includes:

1. Review redirects to the artwork page with a query param
2. Artwork page picks up the query param, renders modal component, and removes query param:
3. Modal component picks up latest offer order
4. Modal component shows up if the latest order's artwork matches with the one on the artwork page.

More comments in the code itself.